src/openvic-simulation/economy/GoodInstance.hpp

```diff
@@ -1,57 +1,20 @@
 #pragma once
 
-#include <memory>
-#include <mutex>
-
 #include "openvic-simulation/economy/GoodDefinition.hpp"
-#include "openvic-simulation/economy/trading/BuyUpToOrder.hpp"
-#include "openvic-simulation/economy/trading/MarketSellOrder.hpp"
-#include "openvic-simulation/misc/GameRulesManager.hpp"
-#include "openvic-simulation/types/fixed_point/FixedPoint.hpp"
+#include "openvic-simulation/economy/trading/GoodMarket.hpp"
 #include "openvic-simulation/types/HasIdentifier.hpp"
 #include "openvic-simulation/types/IdentifierRegistry.hpp"
-#include "openvic-simulation/types/ValueHistory.hpp"
-#include "openvic-simulation/utility/Getters.hpp"
 
 namespace OpenVic {
 	struct GoodInstanceManager;
 
-	struct GoodInstance : HasIdentifierAndColour {
+	struct GoodInstance : HasIdentifierAndColour, GoodMarket {
 		friend struct GoodInstanceManager;
 
 	private:
-		GoodDefinition const& PROPERTY(good_definition);
-
-		static constexpr int32_t exponential_price_change_shift = 7;
-		std::unique_ptr<std::mutex> buy_lock;
-		std::unique_ptr<std::mutex> sell_lock;
-		GameRulesManager const& game_rules_manager;
-		fixed_point_t absolute_maximum_price;
-		fixed_point_t absolute_minimum_price;
-
-		//TODO get from pool instead of storing as field
-		//only used inside execute_orders()
-		std::vector<fixed_point_t> quantity_bought_per_order;
-		std::vector<fixed_point_t> purchasing_power_per_order;
-
-		//only used during day tick (from actors placing order until execute_orders())
-		std::vector<GoodBuyUpToOrder> buy_up_to_orders;
-		std::vector<GoodMarketSellOrder> market_sell_orders;
-
-		fixed_point_t PROPERTY(price);
-		fixed_point_t PROPERTY(price_inverse);
-		fixed_point_t PROPERTY(price_change_yesterday);
-		fixed_point_t PROPERTY(max_next_price);
-		fixed_point_t PROPERTY(min_next_price);
-		bool PROPERTY(is_available);
-		fixed_point_t PROPERTY(total_demand_yesterday);
-		fixed_point_t PROPERTY(total_supply_yesterday);
-		fixed_point_t PROPERTY(quantity_traded_yesterday);
-		ValueHistory<fixed_point_t> PROPERTY(price_history);
+		GoodDefinition const& good_definition;
```

Comment:
```suggestion
		GoodDefinition const& PROPERTY(good_definition);
```
This needs a getter, it's used in main repo UI code, as well as generally being part of the definition-instance architectural design.

Also could you add a newline between the constructor and `public` below here